node/libs/concurrency/src/error.rs

```diff
@@ -0,0 +1,19 @@
+//! Generalization of anyhow::Context to more structured errors.
+use std::fmt::Display;
+
+/// Trait complementary to anyhow::Context which allows for
+/// adding context to error types which contain anyhow::Error.
+pub trait Wrap: Sized {
+    /// Appends context `c` to the error.
+    fn wrap<C: Display + Send + Sync + 'static>(self, c: C) -> Self {
```

Comment:
Bikeshedding: It could be wordy, but I'd slightly prefer to align with `Error` variant naming and `anyhow::Context`:

```rust
// Simplified
pub trait InternalContext {
    fn internal_context<C>(self, context: C) -> Self;
    fn with_internal_context<C, F>(self, context: F) -> Self;
}
```

Reply:
This is very wordy indeed. IMO brevity is the main purpose of having the Context trait in the first place - otherwise you could argue that Result shouldn't implement Context, because `result.map_err(|err|err.context("..."))` is good enough.